src/index.ts

```diff
@@ -149,11 +191,30 @@ export { extractPublicKey } from './utils.js'
  */
 const signLegacyV1 = async (privateKey: PrivateKey, value: Uint8Array, validityType: IpnsEntry.ValidityType, validity: Uint8Array): Promise<Uint8Array> => {
   try {
-    const dataForSignature = ipnsEntryDataForV1Sig(value, validityType, validity)
+    const dataForSignature = ipnsRecordDataForV1Sig(value, validityType, validity)
 
     return await privateKey.sign(dataForSignature)
   } catch (error: any) {
     log.error('record signature creation failed', error)
     throw errCode(new Error('record signature creation failed'), ERRORS.ERR_SIGNATURE_CREATION)
   }
 }
+
+/**
+ * Normalizes the given record value. It ensures it is a string starting with '/'.
+ * If the given value is a cid, the returned path will be '/ipfs/{cid}'.
+ */
+const normalizeValue = (value: string | Uint8Array): string => {
+  const str = typeof value === 'string' ? value : uint8ArrayToString(value)
+
+  if (str.startsWith('/')) {
+    return str
+  }
+
+  try {
+    const cid = CID.parse(str)
```

Comment:
Parsing the CID like this is not free - if we want to accept CIDs as an argument, we should accept a CID object instead of trying to parse it from a string.

Reply:
Again, we don't want to. But that's what's already happening (no validation of what value was whatsoever): https://github.com/ipfs/js-ipns/pull/234#discussion_r1304304990